contracts/src/v0.8/dev/BatchVRFCoordinatorV2.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "./VRFTypes.sol";
+
+/**
+ * @title BatchVRFCoordinatorV2
+ * @notice The BatchVRFCoordinatorV2 contract acts as a proxy to write many random responses to the
+ *   provided VRFCoordinatorV2 contract efficiently in a single transaction.
+ */
+contract BatchVRFCoordinatorV2 {
+  VRFCoordinatorV2 public immutable COORDINATOR;
+
+  event ErrorReturned(string reason);
+  event RawErrorReturned(bytes lowLevelData);
```

Comment:
Would it be worth adding extra info to these events, e.g request ID? We can calculate the request ID using this from `VRFCoordinatorV2`:

```solidity
bytes32 keyHash = hashOfKey(proof.pk);
uint256 requestId = uint256(keccak256(abi.encode(keyHash, proof.seed)));
```

But we will have to do this after catching within each iteration because we could technically have different keyhashes within the same batch of fulfillments. Not sure how much gas it would cost, but the information might be pretty useful.

Reply:
yeah worth it I'd say, will be super cheap since you already have the proof in memory so no storage reads